From 8c43f4fa381a8f9e19506b6fd9a1316eec53ed11 Mon Sep 17 00:00:00 2001
Message-ID: <8c43f4fa381a8f9e19506b6fd9a1316eec53ed11.1742659897.git.mjg@fedoraproject.org>
From: Michael J Gruber <mjg@fedoraproject.org>
Date: Sat, 22 Mar 2025 10:45:47 +0100
Subject: [PATCH] Work around pip 25/pyproject_hooks 1.2.0 path meddling

pip 25/pyproject_hooks 1.2.0 inserts `_BackendPathFinder` into
`sys.meta_path` at index 0, and that makes the import machinery find the
directory `platform` as a namespace instead of importing the standrd
platform module.

Work around by removing this finder from `sys.meta_path` if it is
present at index 0.

Upstream bug: https://github.com/pypa/pip/issues/13289
---
 setup.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 70440190c..9e2c5f49c 100755
--- a/setup.py
+++ b/setup.py
@@ -33,11 +33,15 @@ Internal testing only - environmental variables:
         run swig.
 '''
 
+# Work around pip 25/pyproject_hooks 1.2.0 path meddling:
+import sys
+if sys.meta_path[0].__class__.__name__ == '_BackendPathFinder':
+    sys.meta_path.pop(0)
+
 import os
 import platform
 import re
 import subprocess
-import sys
 import time
 
 
-- 
2.49.0.488.g4ad1d75fa3

